src/OrbitQt/CaptureOptionsDialog.cpp

```diff
@@ -77,4 +79,36 @@ void CaptureOptionsDialog::ShowSourcePathsMappingEditor() {
   }
 }
 
+void CaptureOptionsDialog::SetCollectMemoryInfo(bool collect_memory_info) {
+  ui_->collectMemoryInfoCheckBox->setChecked(collect_memory_info);
+}
+
+bool CaptureOptionsDialog::GetCollectMemoryInfo() const {
+  return ui_->collectMemoryInfoCheckBox->isChecked();
+}
+
+void CaptureOptionsDialog::SetMemorySamplingPeriodNs(uint64_t memory_sampling_period_ns) {
+  ui_->memorySamplingPeriodNsLineEdit->setText(QString::number(memory_sampling_period_ns));
+}
+
+uint64_t CaptureOptionsDialog::GetMemorySamplingPeriodNs() const {
+  CHECK(!ui_->memorySamplingPeriodNsLineEdit->text().isEmpty());
```

Comment:
This does not work, it crashes when the user leaves the field empty

Reply:
Thanks for pointing out this issue.
I have added the `ResetMemorySamplingPeriodMsLineEditWhenEmpty` method such that the value will be set to the default value (i.e., 100ms) if empty.